# HG changeset patch
# User Darren Salt <linux@youmustbejoking.demon.co.uk>
# Date 1221306490 -3600
# Node ID f5ff69fc65d3229ac9908b13c5bb0b9317adf80c
# Parent  89c3e23896babb1fc6fe0f4299fa645a47297d4c
Compilation fixes for ffmpeg API & ABI changes.

diff -r 89c3e23896ba -r f5ff69fc65d3 src/combined/ffmpeg/ff_audio_decoder.c
--- a/src/combined/ffmpeg/ff_audio_decoder.c	Fri Sep 05 21:26:59 2008 +0200
+++ b/src/combined/ffmpeg/ff_audio_decoder.c	Sat Sep 13 12:48:10 2008 +0100
@@ -322,7 +322,7 @@
 
     if (!this->output_open) {
       if (!this->audio_bits || !this->audio_sample_rate || !this->audio_channels) {
-        avcodec_decode_audio (this->context,
+        avcodec_decode_audio2 (this->context,
                               (int16_t *)this->decode_buffer,
                               &decode_buffer_size,
                               &this->buf[0],
diff -r 89c3e23896ba -r f5ff69fc65d3 src/combined/ffmpeg/ffmpeg_decoder.h
--- a/src/combined/ffmpeg/ffmpeg_decoder.h	Fri Sep 05 21:26:59 2008 +0200
+++ b/src/combined/ffmpeg/ffmpeg_decoder.h	Sat Sep 13 12:48:10 2008 +0100
@@ -33,6 +33,10 @@
 #  include "../../libffmpeg/libavcodec/avcodec.h"
 #endif
 
+#if LIBAVCODEC_VERSION_MAJOR > 51
+#define bits_per_sample bits_per_coded_sample
+#endif
+
 typedef struct ff_codec_s {
   uint32_t          type;
   enum CodecID      id;
